### PR TITLE
feat: use custom formats when `textRenderer` returns `false`

### PR DESCRIPTION
**Pull request by amigian74**
Added config object to textRenderer function

### CODE REVIEW

**Comment by robertsLando** [comment]
IMO this is not needed, when you create a new Justgage instance you should be able to access the config from the instance. Also how do you handle the humanNumbers and format functions? Them should be added to Justgage prototype

**Comment by amigian74** [comment]
Yes you're right - did'nt see this before. How about another approach where the textRenderer function returns false if it cannot format the value. Then the internal functions could take over this work

**Comment by robertsLando** [comment]
I'm ok withh that, in that case you could add a check in the if `textRenderer(value) !== false`

**Comment by robertsLando** [comment]
Also remember to add this to docs in readme please :)

**Comment by amigian74** [comment]
Maybe it's not the most elegant way, but should work for most of the used cases. In addition it would make sense to 'highlight' non numeric values in the gauge itself - but I'm not sure how to accomplish that.

File: docs/justgage.js [hunk 1]
```diff
@@ -723,36 +723,45 @@
     defs, svg = null;
 
     // set value to display
+    // set original value to false before trying to call textRenderer for later check
+    obj.originalValue = false;
     if (obj.config.textRenderer) {
```
Comment (robertsLando) [inline docs/justgage.js:728]: I would prefer a cleaner approach, like

`if (obj.config.textRenderer && obj.config.textRenderer(obj.originalValue) !== false)`

File: README.md [hunk 2]
```diff
@@ -93,56 +93,56 @@ setInterval(() => {
...
+| min                  | `0`                                 | Min value                                                                                  |
+| minTxt               | `false`                             | Min value text, overrides `min` if specified                                               |
+| max                  | `100`                               | Max value                                                                                  |
+| maxTxt               | `false`                             | Max value text, overrides `max` if specified                                               |
+| reverse              | `false`                             | Reverse min and max                                                                        |
+| humanFriendlyDecimal | `0`                                 | Number of decimal places for our human friendly number to contain                          |
+| textRenderer         | `null`                              | Function applied before redering text `(value) => value` return false for common behaviour |
```
Comment (robertsLando) [inline README.md:113]:
```suggestion
| textRenderer         | `null`                              | Function applied before redering text `(value: number) => value` return `false` to format value based on config options |
```

**Review by robertsLando — CHANGES_REQUESTED** [review]

File: docs/justgage.js (hunk 1 above)
Comment (amigian74) [reply docs/justgage.js]: done

File: README.md (hunk 2 above)
Comment (amigian74) [reply README.md]: done

**Comment by amigian74** [comment]
What change requests do you have? I can't see any

File: justgage.js [hunk 3]
```diff
@@ -724,7 +724,7 @@
 
     // set value to display
     if (obj.config.textRenderer) {
-      obj.originalValue = obj.config.textRenderer(obj.originalValue);
+      obj.originalValue = obj.config.textRenderer(obj.originalValue, obj.config);
```
Comment (robertsLando) [inline justgage.js:727]: ATM this is the main justgage file, the other one is a copy of this and is used on docs. You have something missing in both files. For example in this you are missing the return value check and in the other you are missing the config passed to the textrender function

**Review by robertsLando — CHANGES_REQUESTED** [review]

File: justgage.js (hunk 3 above)
Comment (amigian74) [reply justgage.js]: Sorry - easy to overlook

**Review by robertsLando — APPROVED** [review]

**Pull request merged**